go/private/actions/archive.bzl

```diff
@@ -129,12 +138,28 @@ def emit_archive(go, source = None, _recompile_suffix = "", recompile_internal_d
             archives = direct,
             out_lib = out_lib,
             out_export = out_export,
+            out_nogo_log = out_nogo_log,
             gc_goopts = source.gc_goopts,
             cgo = False,
             testfilter = testfilter,
             recompile_internal_deps = recompile_internal_deps,
         )
 
+    if go.nogo:
+        validations.append(out_nogo_validation)
+        go.actions.run_shell(
+            inputs = [out_nogo_log],
+            outputs = [out_nogo_validation],
+            # Create the expected output file if there are no errors, otherwise print the errors to stderr.
+            command = '[ -s "$1" ] && cat <(echo) "$1" <(echo) 1>&2 || touch $2',
+            arguments = [
+                out_nogo_log.path,
+                out_nogo_validation.path,
+            ],
+            progress_message = "Nogo Static Analysis",
```

Comment:
This is slightly misleading as the action just processes the results, it doesn't analyze anything. 

Also, purely from a stylistic point of view, progress messages typically start with a verb in progressive form, e.g. `Post-processing nogo findings`.